### ISSUE TITLE
Firefox Quantum

### ISSUE THREAD

**Issue by andrewducker** [mjg/LocalLink#2]
Is there an issue on Firefox Bugzilla for the lacking functionality?

Chrome has the LocalLink addon, so Firefox should support something similar for parity.

**Comment by mjg** [comment]
So far I have not found the necessary support in Firefox. But thanks for pointing out the Chrome addon, I'll check whether its methods can be carried over.

**Comment by bradstech01** [comment]
Just throwing my 2 cents in here because I was checking on whether this had already been done today - I've taken a cursory look at the LocalLinks chrome add on and thrown some effort at trying to port it back to firefox. It looks like it would depend pretty heavily on the chrome.tabs.update and chrome.tabs.create methods (or the "browser" namespace equivalents for firefox). The problem is, if you try to update a tab's URL to a file:// link or create a new tab with a file:// link using those API, you'll get an error like this:

_Unchecked lastError value: Error: Illegal URL: file:///D:/Programming/Web/test.pdf_

It would seem that the moz-extension:// URI doesn't have the same permission as the chrome:// URI had in this regard, so unless that changes, I think it would take some left field solution to get it working.

**Comment by mjg** [comment]
Yes, I arrived at the same conclusion, and I've pretty much given up on this, sorry. (In fact, I've given up on everything Mozilla, but that's a different issue.)
